uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChange.php

```diff
@@ -687,8 +658,8 @@ private static function getDatesOverlappingBalanceChangesToExpire($balanceChange
 
       $wherePeriodEntitlementDates = [];
       $periodStartAndEndDates = $periodEntitlement->getStartAndEndDates();
-      foreach($periodStartAndEndDates as $dates) {
-        $wherePeriodEntitlementDates[] = "leave_request_date.date BETWEEN '{$dates['start_date']}' AND '{$dates['end_date']}'";
+      foreach($periodStartAndEndDates as $date) {
```

Comment:
I had an issue where the $dates array here that conflicting with the $dates array returned for the main method in one of the tests. I'll rename the other to something suitable.